sigmastate/src/main/scala/sigmastate/serialization/ContractTemplateJsonEncoder.scala

```diff
@@ -0,0 +1,71 @@
+package sigmastate.serialization
+
+import io.circe._
+import io.circe.syntax._
+import org.ergoplatform.ErgoBox
+import org.ergoplatform.ErgoBox.NonMandatoryRegisterId
+import org.ergoplatform.settings.ErgoAlgos
+import scalan.RType
+import scorex.crypto.hash.Digest32
+import scorex.util._
+import sigmastate.Values.{Constant, ContractTemplate, EvaluatedValue, Parameter}
+import sigmastate._
+import sigmastate.eval._
+import sigmastate.lang.SigmaParser
+import sigmastate.lang.exceptions.SerializerException
+import special.collection.{Coll, collRType}
+import special.sigma._
+import spire.syntax.all.cfor
+
+import java.math.BigInteger
+import scala.collection.mutable
+
+object ContractTemplateJsonEncoder {
+  def encode(contractTemplate: ContractTemplate): Json = {
+    val obj = mutable.MutableList.empty[(String, Json)]
+
+    obj += ("name" -> Json.fromString(contractTemplate.name))
+    obj += ("description" -> Json.fromString(contractTemplate.description))
+    obj += ("constTypes" -> contractTemplate.constTypes.map(tpe => tpe.toTermString).asJson)
+    obj += ("parameters" -> contractTemplate.parameters.map(param => encodeParameter(param)).asJson)
+
+    contractTemplate.treeVersion match {
+      case Some(version) => obj += ("treeVersion" -> version.asJson)
+    }
+
+    contractTemplate.constValues match {
+      case Some(values) =>
+        obj += ("constValues" -> values.indices.map(i => values(i) match {
+          case Some(const) => DataJsonEncoder.encode(const, contractTemplate.constTypes(i))
+          case None => Json.Null
+        }).asJson)
+    }
+
+    Json.fromFields(obj)
+  }
+
+  private def encodeParameter(param: Parameter): Json = {
+    Json.obj(
+      "name" -> Json.fromString(param.name),
+      "description" -> Json.fromString(param.description),
+      "placeholder" -> Json.fromInt(param.placeholder)
+    )
+  }
+
+  def decode(json: Json): Unit = {
+    val treeVersionOpt = {
+      val treeVersionJson = json.hcursor.downField("treeVersion").focus.get
+      if (treeVersionJson != Json.Null) {
+        Some(treeVersionJson.asNumber.get.toByte.get)
+      } else {
+        None
+      }
+    }
+    val name = json.hcursor.downField("name").focus.get.asString.get
+    val description = json.hcursor.downField("description").focus.get.asString.get
+
+    val tpe = SigmaParser.parseType(json.hcursor.downField("type").focus.get.asString.get)
+
+//    ContractTemplate(treeVersionOpt, name, description)
+  }
+}
```

Comment:
please use JsonCodecs as examples (which is based on circe).
You can use some helpers from ther at well.
But don't add this new code there, instead define it in ContractTemplate companion object.

Reply:
Done.